src/test/scala/com/rawlabs/das/postgresql/DASPostgresqlTable.scala

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2024 RAW Labs S.A.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0, included in the file
+ * licenses/APL.txt.
+ */
+
+package com.rawlabs.das.postgresql
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+import com.rawlabs.das.sdk.DASExecuteResult
+import com.rawlabs.das.sdk.scala.DASTable
+import com.rawlabs.das.sdk.scala.DASTable.TableEstimate
+import com.rawlabs.protocol.das.v1.query._
+import com.rawlabs.protocol.das.v1.tables.TableDefinition
+import com.rawlabs.protocol.das.v1.types.Value
+import com.rawlabs.protocol.das.v1.types.Value.ValueCase
+
+class DASPostgresqlTable(backend: PostgresqlBackend, defn: TableDefinition) extends DASTable {
+
+  /**
+   * Estimate the size of the table.
+   *
+   * @param quals filters applied
+   * @param columns columns to return
+   * @return (expected_number_of_rows (long), avg_row_width (in bytes))
+   */
+  override def tableEstimate(quals: Seq[Qual], columns: Seq[String]): TableEstimate = {
+    // 1) Build the same WHERE clause used in `execute(...)`.
+    val whereClause =
+      if (quals.isEmpty) ""
+      else " WHERE " + quals.map(qualToSql).mkString(" AND ")
+
+    // 2) Possibly use columns if you want to estimate only the subset of columns,
+    // or just use `*` to get the overall row width estimate from the DB.
+    val selectClause =
+      if (columns.isEmpty) "1"
+      else columns.map(quoteIdentifier).mkString(", ")
+
+    val tableName = quoteIdentifier(defn.getTableId.getName)
+    val sql = s"SELECT $selectClause FROM $tableName$whereClause"
+    backend.estimate(sql)
+  }
+
+  /**
+   * Execute a query operation on a table.
+   *
+   * @param quals filters applied
+   * @param columns columns to return
+   * @param sortKeys sort keys to apply
+   * @return a closeable iterator of rows
+   */
+  override def execute(quals: Seq[Qual], columns: Seq[String], sortKeys: Seq[SortKey]): DASExecuteResult = {
+
+    // Build SELECT list
+    val selectClause =
+      if (columns.isEmpty) "*"
+      else columns.map(c => quoteIdentifier(c)).mkString(", ")
+
+    // Build WHERE from `quals`
+    val whereClause =
+      if (quals.isEmpty) ""
+      else " WHERE " + quals.map(qualToSql).mkString(" AND ")
+
+    // Build ORDER BY
+    val orderByClause =
+      if (sortKeys.isEmpty) ""
+      else " ORDER BY " + sortKeys.map(sortKeyToSql).mkString(", ")
+
+    val tableName = quoteIdentifier(defn.getTableId.getName)
+    val sql = s"SELECT $selectClause FROM $tableName$whereClause$orderByClause"
+
+    println(s"Executing SQL: $sql")
+
+    backend.execute(sql)
+  }
+
+  private def quoteIdentifier(ident: String): String = {
+    // naive approach
+    s""""$ident""""
```

Comment:
Yes, needs a regex and adding quotes.

Reply:
I fixed a number of things in PostgresDAS (like it was ignoring the schema), and about the ident, we wrap it in double quotes because we're internally using the postgres identifier, the one that can be wrapped in double quotes. I added the logic to escape double quotes if ever they're part of the identifier.

Fields and tables are advertised such that double quotes is safe. Clients are sending idenfiers that follow those (DAS protocol isn't case insensitive). I think that works like that.